src/System.ComponentModel.TypeConverter/tests/BindingListTests.cs

```diff
@@ -0,0 +1,699 @@
+						OnPropertyChanged ("Name");
+					}
+				}
+			}
+
+			void OnPropertyChanged (string name)
+			{
+				var fn = PropertyChanged;
+				if (fn != null)
+					fn (this, new PropertyChangedEventArgs (name));
+			}
+		}
+
+		// [Fact] // https://bugzilla.xamarin.com/show_bug.cgi?id=8366
+		// public void Bug8366 ()
+		// {
+		// 	bool added = false;
+		// 	bool changed = false;
+		// 	var list = new BindingList<Item> ();
+		// 	list.ListChanged += (object sender, ListChangedEventArgs e) => {
+		// 		added |= e.ListChangedType == ListChangedType.ItemAdded;
+		// 		changed |= e.ListChangedType == ListChangedType.ItemChanged;
+		// 	};
+
+		// 	var item = new Item () { Name = "1" };
+		// 	list.Add (item);
+
+		// 	item.Name = "2";
+
+		// 	Assert.True (added, "ItemAdded");
+		// 	Assert.True (changed, "ItemChanged");
+		// }
+		
+		[Fact] // https://bugzilla.xamarin.com/show_bug.cgi?id=16902
+		public void Bug16902 ()
+		{
+			var list = new BindingList<Item> ();
+			list.Insert (0, null);
+			var count = list.Count;
+			
+			Assert.Equal (1, count);
+		}
+
+		private class Person : INotifyPropertyChanged
+		{
+			private string _lastName;
+			private string _firstName;
+
+			public string FirstName {
+				get { return _firstName; }
+				set {
+					_firstName = value;
+					OnPropertyChanged ("FirstName"); // string matches property name
+				}
+			}
+
+			public string LastName {
+				get { return _lastName; }
+				set {
+					_lastName = value;
+					OnPropertyChanged ("Apepe"); // string doesn't match property name
+				}
+			}
+
+			public event PropertyChangedEventHandler PropertyChanged;
+
+			protected virtual void OnPropertyChanged (string propertyName = null)
+			{
+				PropertyChangedEventHandler handler = PropertyChanged;
+				if (handler != null)
+					handler (this, new PropertyChangedEventArgs (propertyName));
+			}
+		}
+
+		// [Fact] // https://bugzilla.xamarin.com/show_bug.cgi?id=20672
+		// public void Bug20672 ()
+		// {
+		// 	string changedPropertyName = string.Empty;
+		// 	bool isEventRaised = false;
+		// 	bool? hasPropertyDescriptor = false;
+
+		// 	var persons = new BindingList<Person>();
+		// 	persons.Add (new Person() { FirstName = "Stefaan", LastName = "de Vogelaere" });
+		// 	persons.Add (new Person() { FirstName = "Christophe", LastName = "De Langhe" });
+		// 	persons.ListChanged += (object sender, ListChangedEventArgs e) => {
+		// 	    isEventRaised = true;
+		// 	    hasPropertyDescriptor = e.PropertyDescriptor != null;
+		// 	};
+
+		// 	//if the OnPropertyChanged string matches a valid property name, PropertyDescriptor should be generated
+		// 	persons[0].FirstName = "Stefan";
+		// 	Assert.True (isEventRaised);
+		// 	Assert.True ((bool) hasPropertyDescriptor, "#1");
+
+		// 	//if the OnPropertyChanged string doesn't match a valid property name, no PropertyDescriptor should be generated
+		// 	persons[0].LastName = "de le Vulu";
+		// 	Assert.False ((bool) hasPropertyDescriptor, "#2");
+		// }
+	}
+}
```

Comment:
Same questions about this commented out test.  We shouldn't check it in like this.